back-end/API/app/repositories/study_program_repo.py

```
@@ -2,29 +2,36 @@
         studyPrograms = StudyProgram.query.all()
-        studyProgramsJson = [studyProgram.to_json_string() for studyProgram in studyPrograms]
-        study_programs_sorted = sorted(studyProgramsJson, key=lambda e: locale.strxfrm(e['name']))
-        return study_programs_sorted
+        # Fix pas très élégant pour mettre "Tous les programmes" en premier
+        study_programs_sorted = sorted(studyPrograms, key=lambda e: (0 if e.name == "Tous les programmes" else 1, locale.strxfrm(e.name)))
+        dtos = [StudyProgramReadDTO.model_validate(sp) for sp in study_programs_sorted]
+        return dtos
 
-    def findById(self, id):
-        return StudyProgram.query.filter_by(id=id).first()
+    def findById(self, id) -> StudyProgramReadDTO:
```

Ruff (0.15.12)

[error] Function argument `id` is shadowing a Python builtin

(A002)